state/state_test.go

```diff
@@ -2631,13 +2651,46 @@ func (s *StateSuite) TestRemoveExportingModelDocsExporting(c *gc.C) {
 	err = model.SetMigrationMode(state.MigrationModeExporting)
 	c.Assert(err, jc.ErrorIsNil)
 
+	// Logs that should be removed.
+	writeLogs(c, st, 5)
+	// Logs that shouldn't.
+	writeLogs(c, s.State, 5)
+
 	err = st.RemoveExportingModelDocs()
 	c.Assert(err, jc.ErrorIsNil)
 
 	// test that we can not find the user:envName unique index
 	s.checkUserModelNameExists(c, checkUserModelNameArgs{st: st, id: userModelKey, exists: false})
 	s.AssertModelDeleted(c, st)
 	c.Assert(state.HostedModelCount(c, s.State), gc.Equals, 0)
+
+	// These are still there.
+	assertLogCount(c, s.State, 5)
```

Comment:
Gah, I had them as three separate tests first and then decided that was silly for reasons that I can't really articulate. I'll split them out again. I don't mind testing it just for one of them, although it seemed a bit arbitrary which one.